sympy/core/tests/test_relational.py

```diff
@@ -958,6 +958,14 @@ def test_issues_13081_12583_12534():
     # this should be the same if we reverse the relational
     assert [i for i in range(15, 50) if pi.n(i) < Rational(pi.n(i))] == []
 
+def test_issue_18188():
+    from sympy.sets.conditionset import ConditionSet
+    result1 = Eq(x*cos(x) - 3*sin(x), 0)
+    assert isinstance(result1.as_set(), ConditionSet)
+    assert result1.as_set() == ConditionSet(x, Eq(x*cos(x) - 3*sin(x), 0), Reals)
```

Comment:
You're calling `as_set` twice here. It only needs to be called once. Probably the `isinstance` check can just be removed since the `==` below would fail otherwise.

Reply:
Okay, I will remove that.